framework/audio/devtools/audioenginedevtools.cpp

```diff
@@ -99,24 +99,41 @@ void AudioEngineDevTools::makeArpeggio()
 
     m_midiStream = std::make_shared<midi::MidiStream>();
 
-    auto makeEvents = [](Events& events, uint32_t tick, int pitch) {
-                          /* notes of the arpeggio */
-                          static std::vector<int> notes = { 60, 64, 67, 72, 76, 79, 84, 79, 76, 72, 67, 64 };
-                          static uint32_t duration = 4440;
-
-                          uint32_t note_duration = duration / notes.size();
-                          uint32_t note_time = tick;
-
-                          for (int n : notes) {
-                              events.insert({ note_time, Event(0, EventType::ME_NOTEON, n + pitch, 100) });
-                              note_time += note_duration;
-                              events.insert({ note_time, Event(0, EventType::ME_NOTEOFF, n + pitch, 100) });
-                          }
-                      };
-
-    makeEvents(m_midiStream->initData.events, 0, 0);
-
-    m_midiStream->request.onReceive(this, [this, makeEvents](uint32_t tick) {
+    Track t;
+    t.num = 0;
+    t.channels.push_back(0);
+    m_midiStream->initData.tracks.push_back(t);
+
+    midi::Event e;
+    e.channel = 0;
+    e.type = EventType::ME_PROGRAM;
+    e.a = 0;
```

Comment:
It's better to use `Event(channel_t ch, EventType type, int a, int b)`
Event::a and Event::b goes to be private